### PR TITLE
[TF FE] Support Case operation for TensorFlow models

### LINKED ISSUE #20534: [Good First Issue][TF FE]: Support Case operation for TensorFlow models
### Context

OpenVINO component responsible for support of TensorFlow models is called as TensorFlow Frontend (TF FE). TF FE converts a model represented in [TensorFlow opset](https://www.tensorflow.org/api_docs/python/tf/raw_ops) to a model in [OpenVINO opset](https://docs.openvino.ai/2023.1/openvino_docs_ops_opset.html).

In order to infer TensorFlow models with [Case](https://www.tensorflow.org/api_docs/python/tf/raw_ops/Case) operation by OpenVINO, TF FE needs to be extended with this operation support.

### What needs to be done?

For [Case](https://www.tensorflow.org/api_docs/python/tf/raw_ops/Case) operation support, you need to add the corresponding loader into TF FE [op](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/tensorflow_common/src/op) directory and to register it into the dictionary of [Loaders](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/src/op_table.cpp). One loader is responsible for conversion (or decomposition) of one type of TensorFlow operation.

Here is an example of loader implementation for TensorFlow `Einsum` operation:

```cpp
OutputVector translate_einsum_op(const NodeContext& node) { 
     auto op_type = node.get_op_type(); 
     TENSORFLOW_OP_VALIDATION(node, op_type == "Einsum", "Internal error: incorrect usage of translate_einsum_op."); 
     auto equation = node.get_attribute<std::string>("equation"); 
  
     OutputVector inputs; 
     for (size_t input_ind = 0; input_ind < node.get_input_size(); ++input_ind) { 
         inputs.push_back(node.get_input(input_ind)); 
     } 
  
     auto einsum = make_shared<Einsum>(inputs, equation); 
     set_node_name(node.get_name(), einsum); 
     return {einsum}; 
 } 
```

In this example, `translate_einsum_op` converts TF `Einsum` into OV `Einsum`. `NodeContext` object passed into the loader packs all info about inputs and attributes of `Einsum` operation. The loader retrieves an attribute of the equation by using the `NodeContext::get_attribute()` method, prepares input vector, creates `Einsum` operation from OV opset and returns a vector of outputs.

Responsibility of a loader is to parse operation attributes, prepare inputs and express TF operation via OV operations sub-graph. Example for `Einsum` demonstrates the resulted sub-graph with one operation. In PR https://github.com/openvinotoolkit/openvino/pull/19007 you can see operation decomposition into multiple node sub-graph.

### Hint

Case operation is a type of body graph operations that needs special attention to extract and handle body graphs for each case. You can take a look how to approach with body graphs in loaders for While and If operations.

### Example Pull Requests

- https://github.com/openvinotoolkit/openvino/pull/19007

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [OpenVINO TensorFlow Frontend](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/README.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)

### Contact points

@openvinotoolkit/openvino-tf-frontend-maintainers 

### Ticket

TBD

### PR DESCRIPTION
**Pull request by rghvsh**
### Details:
 - *[TF FE] Support Case operation for TensorFlow models*
 - *Closes #20534*

### Tickets:
 - *TBD*

### CODE REVIEW

**Review by rkazants — COMMENTED** [review]
please update `https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/docs/supported_ops.md` with `Case` support. Without it, CI will fail

File: src/frontends/tensorflow/docs/supported_ops.md [hunk 1]
```diff
@@ -191,7 +191,7 @@ A "supported operation" is one that TensorFlow Frontend can convert to the OpenV
 | CTCLossV2                                               | NO                            |                               |
 | CacheDataset                                            | NO                            |                               |
 | CacheDatasetV2                                          | NO                            |                               |
-| Case                                                    | NO                            |                               |
+| Case                                                    | YES                            |                               |
```
Comment (rkazants) [inline src/frontends/tensorflow/docs/supported_ops.md:194]:
```suggestion
| Case                                                    | YES                           |                               |
```

File: src/frontends/tensorflow_common/src/op/case.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "input_model.hpp"
+#include "openvino/opsets/opset10.hpp"
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/case.cpp:7]: please avoid usage of opset header. Use operation header instead like `openvino/op/if.hpp`.

**Comment by rghvsh** [comment]
Hello @rkazants! What changes should I make to fix these issues?

**Comment by rghvsh** [comment]
Hey @rkazants! Will fix the tests

**Comment by rghvsh** [comment]
Hi! @rkazants I don't get the format issue fix suggested here 
image

Thanks 
Raghav

**Comment by mlukasze** [comment]
hey @rghvsh :)
could you please fix conflicts?
we will start review soon (and sorry you had to wait)

**Comment by mlukasze** [comment]
hey [rghvsh](https://github.com/rghvsh) :)
do you want to finish this PR or it's better to move it for another contributor?

**Comment by rghvsh** [comment]
Hi @rkazants was a bit piled up with things can you restart this pr will make the changes 

Thanks 
Raghav

**Pull request closed without merging**